src/controller/java/BaseCHIPCluster-JNI.cpp

```diff
@@ -14,3 +15,46 @@ JNI_METHOD(void, BaseChipCluster, deleteCluster)(JNIEnv * env, jobject self, jlo
         delete cluster;
     }
 }
+
+JNI_METHOD(jobject, BaseChipCluster, getCommandTimeout)
+(JNIEnv * env, jobject self, jlong clusterPtr)
+{
+    chip::DeviceLayer::StackLock lock;
+    chip::Controller::ClusterBase * cluster = reinterpret_cast<chip::Controller::ClusterBase *>(clusterPtr);
+
+    chip::Optional<chip::System::Clock::Timeout> timeout = cluster->GetCommandTimeout();
+    if (!timeout.HasValue())
+    {
+        jobject emptyOptional = nullptr;
+        chip::JniReferences::GetInstance().CreateOptional(nullptr, emptyOptional);
+        return emptyOptional;
+    }
+
+    jobject timeoutOptional = nullptr;
+    jobject timeoutBoxed    = nullptr;
+    jlong timeoutPrimitive  = static_cast<jlong>(timeout.Value().count());
+    chip::JniReferences::GetInstance().CreateBoxedObject("java/lang/Long", "(J)V", timeoutPrimitive, timeoutBoxed);
+    chip::JniReferences::GetInstance().CreateOptional(timeoutBoxed, timeoutOptional);
+    return timeoutOptional;
+}
+
+JNI_METHOD(void, BaseChipCluster, setCommandTimeout)
+(JNIEnv * env, jobject self, jlong clusterPtr, jobject timeoutOptional)
+{
+    chip::DeviceLayer::StackLock lock;
+    chip::Controller::ClusterBase * cluster = reinterpret_cast<chip::Controller::ClusterBase *>(clusterPtr);
+
+    jobject boxedLong = nullptr;
+    chip::JniReferences::GetInstance().GetOptionalValue(timeoutOptional, boxedLong);
+
+    if (boxedLong != nullptr)
+    {
+        jlong timeoutMillis = chip::JniReferences::GetInstance().LongToPrimitive(boxedLong);
+        cluster->SetCommandTimeout(chip::Optional<chip::System::Clock::Timeout>(
```

Comment:
Minor nit: You can use `chip::Optional::MakeOptional(...)` to avoid (re)specifying the type information.